### DISCUSSION TITLE
Check if domain exists

### DISCUSSION THREAD

**Discussion by sinnbeck** [postalserver/postal#2092] [Feature suggestions]
Unsure if this is already built in and I simply am unaware. Let's say a user sends to a non-existent domain. I would assume this would give an instant permanent fail. Instead it just gives a soft-fail and tries to send again several times. And when it finally fails, it will just report that it tried to many times. A "550 Domain not found" error would in this instance be better

I tested sending to foo@djksad.com which i am certain does not exist :) 

If I send to this address using gmail or similar mail providers, it will give an instant permanent fail due to the domain not existing.

**Comment by sinnbeck** [comment]
The same problem occurs on domains that exists but does not have an MX record.

**Comment by willpower232** [comment]
You can see how this is handled here

https://github.com/postalserver/postal/blob/2e7aaaba5a9a37f32bdff9a9fde0379834b2a4db/lib/postal/smtp_sender.rb#L158

I imagine it would be difficult to tell the difference between "this domain does not exist" and "this domain is having a temporary DNS failure" without adding extra code so if you are unable to trust your users to enter correct domain details then you should probably include some extra validation in your system.

**Reply by sinnbeck** [reply]
I see what you mean. I just figured it would fail instantly. I suppose I am just too used to services like mailgun, where it fails right away :)
